core/lib/server/assets.ts

```diff
@@ -12,11 +12,13 @@ export const generateSignedAssetUploadUrl = async (pubId: PubsId, fileName: stri
 	const bucket = env.ASSETS_BUCKET_NAME;
 
 	const client = new S3Client({
+		endpoint: env.ASSETS_STORAGE_ENDPOINT,
 		region: region,
 		credentials: {
 			accessKeyId: key,
 			secretAccessKey: secret,
 		},
+		forcePathStyle: !!env.ASSETS_STORAGE_ENDPOINT, // Required for MinIO
 	});
 	const command = new PutObjectCommand({
 		Bucket: bucket,
```

Comment:
these are all the changes that are necessary to make `minIO` (or any S3 compatible storage solution) work with our existing setup! isn't that great?